src/handlers/next/saleor-webhooks/saleor-webhook.ts

```diff
@@ -60,63 +53,64 @@ export const AsyncWebhookErrorCodeMap: Record<SaleorWebhookError, number> = {
   CONFIGURATION_ERROR: 500,
 };
 
-export type NextWebhookApiHandler<TPayload = unknown, TResp = unknown> = (
+export type NextWebhookApiHandler<TPayload = unknown, TExtras = {}> = (
   req: NextApiRequest,
-  res: NextApiResponse<TResp>,
-  ctx: WebhookContext<TPayload>
+  res: NextApiResponse,
+  ctx: WebhookContext<TPayload> & TExtras
 ) => unknown | Promise<unknown>;
 
-export class SaleorAsyncWebhook<TPayload = unknown> {
+export abstract class SaleorWebhook<
+  TPayload = unknown,
+  TExtras extends Record<string, unknown> = {}
+> {
+  protected abstract type: "async" | "sync";
+
+  // todo breaking type
+  protected extraContext: any = {}; // should be TExtra
```

Comment:
```suggestion
  protected extraContext: unknown = {}; // should be TExtra
```

Why TExtras are not used?

Reply:
It breaks TS code and I dont know how to fix it :( 

Technically its still internal code (its abstract class type, not the exported class). I will have another look into that and if I don't figure it out, I'll leave descriptive comment